android/src/main/java/com/novoda/notils/devicedetection/AmazonDetector.java

```diff
@@ -0,0 +1,17 @@
+package com.novoda.notils.devicedetection;
+
+class AmazonDetector {
+
+    //From: https://developer.amazon.com/appsandservices/solutions/devices/kindle-fire/specifications/01-device-and-feature-specifications
+    private static final String AMAZON = "Amazon";
+
+    private final String buildManufacturer;
+
+    AmazonDetector(String buildManufacturer) {
```

Comment:
>Consider that this class is package protected and not part of the public API.

Yep but since it's going into notils, which is a collection of utility classes, we might not want to couple the classes together. Consider the case where I don't care about x86, and just want an Amazon device detector - the Amazon detector doesn't rely on anything, but I'm forced to create the DeviceDetector, and worse, I'm unable to achieve the second scenario above because I have no way to pass anything in.

>For the second scenario one way to do it would be to pass an array of manufacturers instead of just one value ;)
>
>Also this is in sync with the X86Detector that receives a bunch of values at constructor time that are calculated based on the platform. And we only want to do that once and not every time that we check.

Yep would also argue the same for the x86Detector, the array of supported ABIs should be a method param. The DeviceDetector can still create that list once and not every time we check, but we move that requirement to the deviceDetector, not into these classes.